src/components/ModalWithForm.js

```diff
@@ -25,7 +37,29 @@ export default class ModalWithForm extends Modal {
     });
   }
 
+  // setEventListeners() {
+  //   this._modalForm.addEventListener("submit", (e) => {
+  //     const modalData = this._getInputValues();
+  //     e.preventDefault();
+  //     this.setModalLoad(true);
+  //     this._handleFormSubmit(modalData)
+  //       .then(() => {
+  //         console.log(`Success:`, modalData);
+  //         this.close();
+  //         this._modalForm.reset();
+  //       })
+  //       .catch((err) => {
+  //         console.error(`Error:`, err);
+  //       })
+  //       .finally(() => {
+  //         this.setModalLoad(false);
+  //       });
+  //   });
+  //   super.setEventListeners();
+  // }
+
   close() {
+    this._modalForm.reset();
```

Comment:
You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` or on `close` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.


You should **clear ** the inputs only after **submitting** in `then` a card rather than when you open or close popups because if you unexpectedly close the popup, you will be upset if the inputs are empty when you open it again, and you’ll have to type the data again.